File: code-coaches/medium/duty-free/duty_free.c
#include <stdio.h>
#include <stdlib.h>

int is_on_budget(float *items_price_eur, int num_items)
{
    float usd_eur_rate = 1.1;
    float limit_price_usd = 20;

    for (int i = 0; i < num_items; i++)
    {
        float item_price_eur = items_price_eur[i];
        float item_price_usd = item_price_eur * usd_eur_rate;
        if (item_price_usd > limit_price_usd)
        {
            return 0;
        }
    }

    return 1;
}

int main()
{
    float *items_price_eur = NULL;
    int num_items = 0;
    float price;

    // Read item prices from standard input
    while (scanf("%f", &price) == 1)
    {
        num_items++;
        items_price_eur = (float *)realloc(items_price_eur, num_items * sizeof(float));
        if (items_price_eur == NULL)
        {
            printf("Error: out of memory\n");
            return 1;
        }
        items_price_eur[num_items - 1] = price;
    }

    if (is_on_budget(items_price_eur, num_items))
        printf("On to the terminal\n");
    else
        printf("Back to the store\n");

    // Free memory and return
    free(items_price_eur);
    return 0;
}

File: code-coaches/medium/duty-free/DutyFree.java
import java.util.Scanner;

public class DutyFree {

    public static boolean isOnBudget(double[] itemsPriceEur) {
        double usdEurRate = 1.1;
        double limitPriceUsd = 20;

        for (double itemPriceEur : itemsPriceEur) {
            double itemPriceUsd = itemPriceEur * usdEurRate;
            if (itemPriceUsd > limitPriceUsd) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read item prices from standard input
        double[] itemsPriceEur = new double[100];
        int numItems = 0;
        while (scanner.hasNextDouble()) {
            double price = scanner.nextDouble();
            itemsPriceEur[numItems++] = price;
        }

        // Check if on budget and output result
        if (isOnBudget(itemsPriceEur)) {
            System.out.println("On to the terminal");
        } else {
            System.out.println("Back to the store");
        }

        scanner.close();
    }

}
